src/renderers/dom/client/eventPlugins/SimpleEventPlugin.js

```diff
@@ -483,6 +483,10 @@ for (var type in topLevelEventsToDispatchConfig) {
 var ON_CLICK_KEY = keyOf({onClick: null});
 var onClickListeners = {};
 
+function getDictionaryKey(inst) {
+  return '.' + inst._rootNodeID;
+}
+
```

Comment:
Don't forget, I am as puzzled that this works as you are. 😄 I don't have good explanations, I'm just saying I tried other ways and they didn't help.